server/src/main/java/keywhiz/service/resources/StatusResource.java

```diff
@@ -26,12 +36,50 @@
 public class StatusResource {
   private Environment environment;
   private static final Logger logger = LoggerFactory.getLogger(SecretDeliveryResource.class);
+  LoadingCache<HealthCheckRegistry, SortedMap<String, HealthCheck.Result>> cachedResult;
 
-  @Inject public StatusResource(Environment environment) {
+  @Inject public StatusResource(KeywhizConfig keywhizConfig, Environment environment) {
     this.environment = environment;
+
+    Duration cacheExpiry = keywhizConfig.getStatusCacheExpiry();
+
+    cachedResult = CacheBuilder.newBuilder()
```

Comment:
Also: this class is not a singleton, so the cache/memoization will get lost since the class will likely get reinstantiated on every request.